src/EventStore.Core/Services/Storage/StorageScavenger.cs

```diff
@@ -120,6 +132,17 @@ public class StorageScavenger :
 				}
 			}
 
+			Guid switchChunksLockId;
+			lock (_lock) {
+				switchChunksLockId = _switchChunksLockId;
+			}
+
+			try {
+				_switchChunksLock.TryRelease(switchChunksLockId);
+			} catch (Exception ex) {
+				logger.Error(ex, "SCAVENGING: Unexpected error when releasing the chunk switch lock.");
+			}
+
```

Comment:
I haven't thought about this carefully, but is there a reason to acquire and release the _lock and then acquire it again below? possibly we could just hold the _lock

Reply:
yes, the main reason is for code readability - i think that it makes the critical section(s) much clearer compared to taking the lock on large blocks of code. also it avoids holding the lock for longer than is actually required (although performance's not really critical here)

Reply:
fair enough :+1:

Reply:
> yes, the main reason is for code readability - i think that it makes the critical section(s) much clearer compared to taking the lock on large blocks of code (but there may be a better way). also it avoids holding the lock for longer than is actually required (although performance's not really critical here)

Just as another one. Its not a mine field. What if on a later refactor a blocking operation were to be added to something called inside of the block as I am sure we have all had happen at some point :-D I have seen some really sneaky examples of this ... even things like Console.WriteLine causing a problem ...

Reply:
@gregoryyoung agreed, that's a strong point :+1:

Reply:
@gregoryyoung yeah good point, agreed